### ISSUE TITLE
TextInput widget loses user input text when typing quickly with a thread pool

### ISSUE THREAD

**Issue by ronan-delta** [holoviz/panel#6414]
#### ALL software version info
panel 1.3.8
bokeh 3.3.4

#### Description of expected behavior and the observed behavior
When using the pn.widgets.TextInput widget either alone, as the widget for a param.String, or in the new chat interface I run into an issue where the some of the characters of the input text are lost when the user is typing quickly. 

You can see the text very briefly then it'll look like it's rolled back to a split second before the latest character or two is entered. For example, you will be typing something like "hello world" and all characters will appear then a split second later it will show "hello worl" with the cursor now after the "l". 

#### Complete, minimal, self-contained example code that reproduces the issue
Below is a simple example that appears to replicate the issue on my end, either served locally or on a hosted server. It maybe related to the nthreads parameter as it appears to at least be far less common when that parameter is None/no thread pool is created. 
```
import panel as pn

pn.extension(
    nthreads=0,
)

def dashboard_page():
    return pn.widgets.TextInput(name="Enter text", placeholder="Type here...")

ROUTES = {
    "Dashboard": dashboard_page,
}
pn.serve(ROUTES, port=8501, admin=True, static_dirs={"thumbnails": "./static/thumbnails"})
```

#### Stack traceback and/or browser JavaScript console output
N/A
#### Screenshots or screencasts of the bug in action
Difficult to capture in action, but seems to happen frequently enough when typing fast. 

- [X] I may be interested in making a pull request to address this, but unsure where to start on this one.

**Comment by ahuang11** [comment]
I can reproduce this.

**Comment by chris-branch** [comment]
Same or similar issue with CodeEditor:

https://discourse.holoviz.org/t/codeeditor-widget-is-unusable-when-automatic-theading-is-enabled/7111

**Comment by duarteocarmo** [comment]
Hey @philippjfr  any idea when this will be available? (I know 1.4.5, just wondering if there is a release date for that, or an option to release it sooner?) 

Thanks!
